### PR TITLE
@babel/preset-env flow typings

### PR DESCRIPTION
**Pull request by gorakong**
Test Plan:
- Ran `yarn flow` to verify there are no flow errors

Question: 

packages/utils/babel-preset-transform-runtime/test/babel-plugin-transform-runtime.test.js (in ticket) seems to use @parcel/babel-preset-env instead of @babel/preset-env. Not sure if I should do the same for this file?

### CODE REVIEW

File: packages/transformers/babel/src/env.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// @flow
+// @flow strict-local
 
 import type {Config} from '@parcel/types';
 import type {BabelTargets} from './types';
```
Comment (wbinnssmith) [inline packages/transformers/babel/src/env.js:4]: We can remove the existing definition of `BabelTargets` from this types file and use the new one 😄

**Review by wbinnssmith — COMMENTED** [review]
> packages/utils/babel-preset-transform-runtime/test/babel-plugin-transform-runtime.test.js (in ticket) seems to use @parcel/babel-preset-env instead of @babel/preset-env. Not sure if I should do the same for this file?

Yeah, that's a separate issue. We don't need to address that here.

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.31s | +45.00ms |
| Cached | 1.76s | +14.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 240.00ms | -20.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 239.00ms | +16.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 1.43s | +113.00ms ⚠️ |
| dist/modern/index.7bc22a8b.css | 77.00b | +0.00b | 1.47s | +79.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 26.49s | +226.00ms |
| Cached | 1.69s | +13.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/NotFound.9af20955.js | 532.00b | +0.00b | 1.03s | +98.00ms ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 181.00ms | +46.00ms ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.78m | -4.04s |
| Cached | 4.66s | +37.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.7664a919.js | 2.74mb | -9.00b 🚀 | 1.48m | -3.71s |
| dist/DatePicker.89ba6023.js | 21.07kb | +0.00b | 8.55s | -1.03s 🚀 |
| dist/workerHasher.9dc1181e.js | 11.90kb | +0.00b | 1.48m | +1.29m ⚠️ |
| dist/EmojiTypeAheadComponent.8313d89f.js | 8.61kb | +0.00b | 10.82s | -1.36m 🚀 |
| dist/EmojiUploadComponent.c023b9c0.js | 2.22kb | +0.00b | 10.81s | -40.59s 🚀 |
| dist/date.1de12427.js | 1.88kb | +0.00b | 5.96s | +312.00ms ⚠️ |
| dist/feedback.762993d4.js | 1.78kb | +0.00b | 8.55s | -839.00ms 🚀 |
| dist/workerHasher.4a170deb.js | 1.67kb | +0.00b | 51.20s | +40.06s ⚠️ |
| dist/status.eb990fdd.js | 1.60kb | +0.00b | 6.60s | +581.00ms ⚠️ |
| dist/code.3d01c3f6.js | 1.52kb | +0.00b | 5.96s | +312.00ms ⚠️ |
| dist/16.d03ef2ed.js | 1.51kb | +0.00b | 8.97s | +3.62s ⚠️ |
| dist/16.3d2fa847.js | 1.46kb | +0.00b | 8.93s | +1.21s ⚠️ |
| dist/heading6.e4e9d5d4.js | 1.44kb | +0.00b | 6.75s | +603.00ms ⚠️ |
| dist/heading3.94ce1091.js | 1.43kb | +0.00b | 6.73s | +614.00ms ⚠️ |
| dist/16.039b21a1.js | 1.36kb | +0.00b | 5.50s | -3.59s 🚀 |
| dist/heading5.87b6dae6.js | 1.31kb | +0.00b | 6.75s | +629.00ms ⚠️ |
| dist/expand.983a2c37.js | 1.30kb | +0.00b | 8.55s | -839.00ms 🚀 |
| dist/16.202504a1.js | 1.26kb | +0.00b | 8.93s | +900.00ms ⚠️ |
| dist/heading2.e0e7648d.js | 1.25kb | +0.00b | 6.73s | +614.00ms ⚠️ |
| dist/mention.41dd4317.js | 1.21kb | +0.00b | 6.36s | +471.00ms ⚠️ |
| dist/heading4.926b16d5.js | 1.20kb | +0.00b | 6.73s | +614.00ms ⚠️ |
| dist/divider.6cc8acb0.js | 1.17kb | +0.00b | 5.99s | +338.00ms ⚠️ |
| dist/quote.b58dc9d4.js | 1.16kb | +0.00b | 6.60s | +581.00ms ⚠️ |
| dist/decision.b8aab464.js | 1.13kb | +0.00b | 5.96s | +312.00ms ⚠️ |
| dist/panel-warning.a4dc1737.js | 1.13kb | +0.00b | 6.41s | +493.00ms ⚠️ |
| dist/heading1.e7155c65.js | 1.09kb | +0.00b | 6.64s | +593.00ms ⚠️ |
| dist/panel-error.2d0a3dc5.js | 1.03kb | +0.00b | 6.36s | +471.00ms ⚠️ |
| dist/panel.983b8f36.js | 1.02kb | +0.00b | 6.60s | +581.00ms ⚠️ |
| dist/table.613566d1.js | 1.01kb | +0.00b | 6.64s | +617.00ms ⚠️ |
| dist/panel-success.f52c4b4f.js | 990.00b | +0.00b | 6.41s | +520.00ms ⚠️ |
| dist/panel-note.af0bab14.js | 986.00b | +0.00b | 6.36s | +471.00ms ⚠️ |
| dist/EmojiPickerComponent.57313a93.js | 680.00b | +0.00b | 10.81s | -40.59s 🚀 |
| dist/simpleHasher.948d1b20.js | 673.00b | +0.00b | 1.48m | +1.29m ⚠️ |
| dist/index.html | 119.00b | +0.00b | 7.55s | +1.27s ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.1a36c238.js | 2.71mb | +0.00b | 523.00ms | +112.00ms ⚠️ |
| dist/pdfRenderer.f8165fbc.js | 1.11mb | +6.00b ⚠️ | 734.00ms | +50.00ms ⚠️ |
| dist/editorView.8e3d6b32.js | 586.93kb | +0.00b | 782.00ms | +62.00ms ⚠️ |
| dist/popup.95b6e35c.js | 169.09kb | +0.00b | 766.00ms | +82.00ms ⚠️ |
| dist/Toolbar.b45f0082.js | 92.27kb | +0.00b | 782.00ms | +62.00ms ⚠️ |
| dist/EmojiPickerComponent.ab132fac.js | 75.00kb | +0.00b | 701.00ms | +45.00ms ⚠️ |
| dist/media-viewer.b113ec80.js | 71.78kb | +0.00b | 734.00ms | +50.00ms ⚠️ |
| dist/card.e15cb2dd.js | 54.95kb | +0.00b | 734.00ms | +50.00ms ⚠️ |
| dist/card.3dbd8c2c.js | 52.71kb | +0.00b | 782.00ms | +84.00ms ⚠️ |
| dist/Modal.9a26da87.js | 41.08kb | +0.00b | 542.00ms | +95.00ms ⚠️ |
| dist/component.327022bf.js | 30.85kb | +0.00b | 523.00ms | +112.00ms ⚠️ |
| dist/component.49a28949.js | 22.36kb | +0.00b | 542.00ms | +95.00ms ⚠️ |
| dist/DatePicker.1626269c.js | 20.94kb | +0.00b | 707.00ms | +51.00ms ⚠️ |
| dist/EmojiPickerComponent.7963a811.js | 18.12kb | +0.00b | 701.00ms | +45.00ms ⚠️ |
| dist/dropzone.cfa9b383.js | 17.22kb | +0.00b | 766.00ms | +62.00ms ⚠️ |
| dist/smartMediaEditor.204c7bbc.js | 16.79kb | +0.00b | 782.00ms | +84.00ms ⚠️ |
| dist/js.eb39c3cf.js | 16.55kb | +0.00b | 542.00ms | +95.00ms ⚠️ |
| dist/ui.cfe8a82a.js | 13.94kb | +0.00b | 734.00ms | +69.00ms ⚠️ |
| dist/workerHasher.9dc1181e.js | 11.90kb | +0.00b | 734.00ms | +70.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.8313d89f.js | 8.61kb | +0.00b | 734.00ms | +70.00ms ⚠️ |
| dist/component.fe59914e.js | 6.19kb | +0.00b | 557.00ms | +110.00ms ⚠️ |
| dist/card.7b75c38c.js | 5.67kb | +0.00b | 734.00ms | +50.00ms ⚠️ |
| dist/media-viewer.39298996.js | 4.01kb | +0.00b | 766.00ms | +82.00ms ⚠️ |
| dist/EmojiPickerComponent.d165b446.js | 3.56kb | +0.00b | 701.00ms | +45.00ms ⚠️ |
| dist/png-chunks-extract.3bcd3532.js | 3.55kb | +0.00b | 542.00ms | +95.00ms ⚠️ |
| dist/index.ac562c80.css | 3.46kb | +0.00b | 833.00ms | +63.00ms ⚠️ |
| dist/dropzone.cca9436f.js | 3.32kb | +0.00b | 766.00ms | +62.00ms ⚠️ |
| dist/Modal.725e4b5d.js | 3.07kb | +0.00b | 532.00ms | -111.00ms 🚀 |
| dist/clipboard.dddd75a9.js | 2.83kb | +0.00b | 766.00ms | +62.00ms ⚠️ |
| dist/16.984a552d.js | 2.49kb | +0.00b | 634.00ms | +204.00ms ⚠️ |
| dist/EmojiUploadComponent.c023b9c0.js | 2.22kb | +0.00b | 701.00ms | +45.00ms ⚠️ |
| dist/card.bc6d3fe8.js | 2.15kb | +0.00b | 766.00ms | +82.00ms ⚠️ |
| dist/ResourcedEmojiComponent.4e7708b7.js | 2.07kb | +0.00b | 524.00ms | +113.00ms ⚠️ |
| dist/date.1de12427.js | 1.88kb | +0.00b | 660.00ms | +135.00ms ⚠️ |
| dist/16.51a973ae.js | 1.86kb | +0.00b | 615.00ms | +134.00ms ⚠️ |
| dist/images.bd2c65c2.js | 1.82kb | +0.00b | 661.00ms | +120.00ms ⚠️ |
| dist/16.664f2364.js | 1.79kb | +0.00b | 578.00ms | +113.00ms ⚠️ |
| dist/browser.4322401e.js | 1.70kb | +0.00b | 766.00ms | +62.00ms ⚠️ |
| dist/workerHasher.4a170deb.js | 1.67kb | +0.00b | 734.00ms | +70.00ms ⚠️ |
| dist/workerHasher.c6b3a06a.js | 1.67kb | +0.00b | 782.00ms | +84.00ms ⚠️ |
| dist/list-number.140c0835.js | 1.60kb | +0.00b | 658.00ms | +98.00ms ⚠️ |
| dist/status.eb990fdd.js | 1.60kb | +0.00b | 672.00ms | +88.00ms ⚠️ |
| dist/code.3d01c3f6.js | 1.52kb | +0.00b | 651.00ms | +126.00ms ⚠️ |
| dist/16.d03ef2ed.js | 1.51kb | +0.00b | 610.00ms | +129.00ms ⚠️ |
| dist/16.2560b363.js | 1.46kb | +0.00b | 578.00ms | +113.00ms ⚠️ |
| dist/16.3d2fa847.js | 1.46kb | +0.00b | 592.00ms | +127.00ms ⚠️ |
| dist/link.0f2a27b0.js | 1.44kb | +0.00b | 652.00ms | +111.00ms ⚠️ |
| dist/16.a99d351e.js | 1.44kb | +0.00b | 636.00ms | +128.00ms ⚠️ |
| dist/heading3.94ce1091.js | 1.43kb | +0.00b | 688.00ms | +58.00ms ⚠️ |
| dist/16.c8c5e38d.js | 1.40kb | +0.00b | 636.00ms | +128.00ms ⚠️ |
| dist/emoji.ea46e5f5.js | 1.37kb | +0.00b | 660.00ms | +119.00ms ⚠️ |
| dist/16.039b21a1.js | 1.36kb | +0.00b | 634.00ms | +125.00ms ⚠️ |
| dist/16.8938ddc0.js | 1.32kb | +0.00b | 634.00ms | +126.00ms ⚠️ |
| dist/16.955db3fc.js | 1.31kb | +0.00b | 650.00ms | +126.00ms ⚠️ |
| dist/expand.983a2c37.js | 1.30kb | +0.00b | 707.00ms | +51.00ms ⚠️ |
| dist/16.3d8988cf.js | 1.27kb | +0.00b | 611.00ms | +130.00ms ⚠️ |
| dist/16.b709596e.js | 1.26kb | +0.00b | 578.00ms | +113.00ms ⚠️ |
| dist/16.202504a1.js | 1.26kb | +0.00b | 610.00ms | +129.00ms ⚠️ |
| dist/16.5ba30215.js | 1.26kb | +0.00b | 610.00ms | +129.00ms ⚠️ |
| dist/16.2da93a26.js | 1.26kb | +0.00b | 578.00ms | +113.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.7bdb6beb.js | 1.25kb | +0.00b | 766.00ms | +82.00ms ⚠️ |
| dist/heading2.e0e7648d.js | 1.25kb | +0.00b | 700.00ms | +70.00ms ⚠️ |
| dist/mention.41dd4317.js | 1.21kb | +0.00b | 666.00ms | +106.00ms ⚠️ |
| dist/heading4.926b16d5.js | 1.20kb | +0.00b | 689.00ms | +59.00ms ⚠️ |
| dist/layout.a7a50d60.js | 1.19kb | +0.00b | 652.00ms | +111.00ms ⚠️ |
| dist/16.eb0e686d.js | 1.18kb | +0.00b | 634.00ms | +126.00ms ⚠️ |
| dist/divider.6cc8acb0.js | 1.17kb | +0.00b | 660.00ms | +119.00ms ⚠️ |
| dist/quote.b58dc9d4.js | 1.16kb | +0.00b | 672.00ms | +88.00ms ⚠️ |
| dist/component.f51f4183.js | 1.15kb | +0.00b | 524.00ms | +113.00ms ⚠️ |
| dist/action.c1329039.js | 1.15kb | +0.00b | 651.00ms | +127.00ms ⚠️ |
| dist/decision.b8aab464.js | 1.13kb | +0.00b | 660.00ms | +135.00ms ⚠️ |
| dist/panel-warning.a4dc1737.js | 1.13kb | +0.00b | 672.00ms | +88.00ms ⚠️ |
| dist/list.6772c571.js | 1.10kb | +0.00b | 666.00ms | +106.00ms ⚠️ |
| dist/heading1.e7155c65.js | 1.09kb | +0.00b | 700.00ms | +70.00ms ⚠️ |
| dist/panel-error.2d0a3dc5.js | 1.03kb | +0.00b | 674.00ms | +114.00ms ⚠️ |
| dist/panel.983b8f36.js | 1.02kb | +0.00b | 672.00ms | +88.00ms ⚠️ |
| dist/table.613566d1.js | 1.01kb | +0.00b | 672.00ms | +42.00ms ⚠️ |
| dist/panel-success.f52c4b4f.js | 990.00b | +0.00b | 700.00ms | +116.00ms ⚠️ |
| dist/panel-note.af0bab14.js | 986.00b | +0.00b | 674.00ms | +114.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.debe5415.js | 960.00b | +0.00b | 766.00ms | +62.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.eaabcc04.js | 956.00b | +0.00b | 734.00ms | +50.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.a56d8c27.js | 956.00b | +0.00b | 782.00ms | +84.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.6a26a5d6.js | 876.00b | +0.00b | 766.00ms | +82.00ms ⚠️ |
| dist/EmojiPickerComponent.57313a93.js | 680.00b | +0.00b | 701.00ms | +45.00ms ⚠️ |
| dist/simpleHasher.948d1b20.js | 673.00b | +0.00b | 734.00ms | +69.00ms ⚠️ |
| dist/simpleHasher.2437bfc1.js | 673.00b | +0.00b | 782.00ms | +84.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/281765467555627532)**

**Review by padmaia — APPROVED** [review]

File: flow-libs/babel-preset-env.js.flow [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  declare type assertVersionCb = () => boolean;
+
+  declare export type Targets = {
+    browsers?: string | Array<string>,
+    esmodules?: boolean,
+    [string]: string,
+    ...
+  };
+
+  declare export type PresetEnvPlugins = Array<mixed>;
```
Comment (wbinnssmith) [inline flow-libs/babel-preset-env.js.flow:13]: This is actually a tuple (array of fixed length of 2): https://github.com/babel/babel/blob/e498bee10f0123bb208baa228ce6417542a2c3c4/packages/babel-preset-env/src/index.js#L321

File: flow-libs/babel-preset-env.js.flow [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  declare type assertVersionCb = () => boolean;
+
+  declare export type Targets = {
+    browsers?: string | Array<string>,
+    esmodules?: boolean,
+    [string]: string,
+    ...
+  };
+
+  declare export type PresetEnvPlugins = [mixed, mixed];
```
Comment (wbinnssmith) [inline flow-libs/babel-preset-env.js.flow:13]: Nit: `PresetEnvPlugin`?

Comment (gorakong) [reply flow-libs/babel-preset-env.js.flow]: @wbinnssmith updated the PR!

**Pull request merged**